py/module.go

```diff
@@ -43,45 +113,55 @@ func (m *Module) GetDict() StringDict {
 }
 
 // Define a new module
-func NewModule(name, doc string, methods []*Method, globals StringDict) *Module {
+func (store *Store) NewModule(ctx Ctx, info ModuleInfo, methods []*Method, globals StringDict) *Module {
+	if info.Name == "" {
+		info.Name = "__main__"
+	}
 	m := &Module{
-		Name:    name,
-		Doc:     doc,
-		Globals: globals.Copy(),
+		ModuleInfo: info,
+		Globals:    globals.Copy(),
+		Ctx:        ctx,
 	}
 	// Insert the methods into the module dictionary
+	// Copy each method an insert each "live" with a ptr back to the module (which can also lead us to the host Ctx)
 	for _, method := range methods {
-		m.Globals[method.Name] = method
+		methodInst := new(Method)
+		*methodInst = *method
+		methodInst.Module = m
+		m.Globals[method.Name] = methodInst
 	}
 	// Set some module globals
-	m.Globals["__name__"] = String(name)
-	m.Globals["__doc__"] = String(doc)
+	m.Globals["__name__"] = String(info.Name)
+	m.Globals["__doc__"] = String(info.Doc)
 	m.Globals["__package__"] = None
+	if len(info.FileDesc) > 0 {
+		m.Globals["__file__"] = String(info.FileDesc)
+	}
 	// Register the module
-	modules[name] = m
+	store.modules[info.Name] = m
 	// Make a note of some modules
-	switch name {
+	switch info.Name {
 	case "builtins":
-		Builtins = m
+		store.Builtins = m
 	case "importlib":
-		Importlib = m
+		store.Importlib = m
 	}
 	// fmt.Printf("Registering module %q\n", name)
 	return m
 }
 
 // Gets a module
-func GetModule(name string) (*Module, error) {
-	m, ok := modules[name]
+func (store *Store) GetModule(name string) (*Module, error) {
+	m, ok := store.modules[name]
 	if !ok {
-		return nil, ExceptionNewf(ImportError, "Module %q not found", name)
+		return nil, ExceptionNewf(ImportError, "Module '%q' not found", name)
```

Comment:
```suggestion
		return nil, ExceptionNewf(ImportError, "Module '%s' not found", name)
```
(`'%q'` sounds a bit redundant)